components/mediation-ui/mediators-ui/org.wso2.carbon.mediator.cache.ui/src/main/java/org/wso2/carbon/mediator/cache/ui/CacheMediator.java

```diff
@@ -21,439 +21,483 @@
+    private static final QName ATT_SEQUENCE = new QName(CachingConstants.SEQUENCE_STRING);
+
+    /**
+     * QName of the implementation type
+     */
+    private static final QName ATT_TYPE = new QName(CachingConstants.TYPE_STRING);
+
+    /**
+     * QName of the maximum message size
+     */
+    private static final QName ATT_SIZE = new QName(CachingConstants.MAX_SIZE_STRING);
+
+    /**
+     * QName of the onCacheHit mediator sequence reference
+     */
+    private static final QName ON_CACHE_HIT_Q = new QName(XMLConfigConstants.SYNAPSE_NAMESPACE,
+                                                          CachingConstants.ON_CACHE_HIT_STRING);
+
+    /**
+     * QName of the cache implementation
+     */
+    private static final QName IMPLEMENTATION_Q = new QName(XMLConfigConstants.SYNAPSE_NAMESPACE,
+                                                            CachingConstants.IMPLEMENTATION_STRING);
+
+    /**
+     * This holds the default timeout of the mediator cache
+     */
+    private static final long DEFAULT_TIMEOUT = 5000L;
+
+    /**
+     * QName of the onCacheHit mediator sequence reference
+     */
+    private static final QName PROTOCOL_Q = new QName(XMLConfigConstants.SYNAPSE_NAMESPACE,
+                                                      CachingConstants.PROTOCOL_STRING);
+
+    /**
+     * QName of the hTTPMethodToCache
+     */
+    private static final QName HTTP_METHODS_TO_CACHE_Q = new QName(XMLConfigConstants.SYNAPSE_NAMESPACE,
+                                                                   CachingConstants.METHODS_STRING);
+
+    /**
+     * QName of the headersToExcludeInHash
+     */
+    private static final QName HEADERS_TO_EXCLUDE_IN_HASH_Q = new QName(XMLConfigConstants.SYNAPSE_NAMESPACE,
+                                                                        CachingConstants.HEADERS_TO_EXCLUDE_STRING);
+    /**
+     * QName of the response codes to include when hashing
+     */
+    private static final QName RESPONSE_CODES_Q = new QName(XMLConfigConstants.SYNAPSE_NAMESPACE,
+                                                            CachingConstants.RESPONSE_CODES_STRING);
+
+    /**
+     * QName of the digest generator
+     */
+    private static final QName HASH_GENERATOR_Q = new QName(XMLConfigConstants.SYNAPSE_NAMESPACE,
+                                                            CachingConstants.HASH_GENERATOR_STRING);
+
+    /**
+     * This specifies whether the mediator should be in the incoming path (to check the request) or in the outgoing path
+     * (to cache the response).
+     */
+    private boolean collector = false;
+
+    /**
+     * This is used to define the logic used by the mediator to evaluate the hash values of incoming messages.
+     */
+    private String digestGenerator = CachingConstants.DEFAULT_HASH_GENERATOR.getClass().toString();
+
+    /**
+     * The size of the messages to be cached in memory. If this is 0 then no disk cache, and if there is no size
```

Comment:
Is it inMemoryCacheSize of disk cach size, also the comment says that 0 is no cache. Shall we correct this?

Reply:
Fixed in #890